socorro/external/boto/connection_context.py

```diff
@@ -14,20 +14,78 @@
 from configman import Namespace, RequiredConfig, class_converter
 
 from socorrolib.lib.converters import change_default
+from socorrolib.lib.ooid import dateFromOoid
 
 
 class KeyNotFound(Exception):
     pass
 
 
 class JSONISOEncoder(json.JSONEncoder):
-
     def default(self, obj):
         if isinstance(obj, datetime.date):
             return obj.isoformat()
         raise NotImplementedError("Don't know about {0!r}".format(obj))
 
 
+class KeyBuilderBase(object):
+    """Base key builder for s3 pseudo-filenames"""
+    def build_keys(self, prefix, name_of_thing, id):
+        """
+        Use S3 pseudo-directories to make it easier to list/expire later
+        {prefix}/v1/{name_of_thing}/{id}
+        """
+        return [
+            '%s/v1/%s/%s' % (prefix, name_of_thing, id)
+        ]
+
+
+class DatePrefixKeyBuilder(KeyBuilderBase):
```

Comment:
Whenever I read that I ready "dateprefix key builder" :)
It's note a dateprefix. It's "date key builder with prefix" or something.